### PR TITLE
Reorder boot/code spec includes for retail

### PR DESCRIPTION
**Pull request by cadmic**
Unfortunately this makes things a bit of a mess. I'm not sure if the original devs' tooling did this automatically.

For functions that are no longer linked, I chose not to `#if` the declarations. The function definitions themselves are not `#if`'d out, so a linker error seems more appropriate if you use one of these functions without updating the spec.

I made some temporary changes to camera.c to avoid referencing unlinked functions. These will have to be cleaned up for real when we decomp those files.

Moved from `boot` to `code`:
```
ldiv
sprintf
string
vimodefpallan1
xldtob
xlitob
xprintf
```
Moved around within `code`:
```
contpfs
pfsgetstatus
pfsisplug
```
No longer linked:
```
assert
db_camera
debug_malloc
epiwrite
mempak
mtxl2f
mtxuty-cvt
pfsallocatefile
pfschecker
pfsdeletefile
pfsfilestate
pfsfindfile
pfsfreeblocks
pfsinitpak
pfsreadwritefile
ucode_disas
z_text
```

### CODE REVIEW

File: src/code/z_camera.c [hunk 1]
```diff
@@ -7277,6 +7281,7 @@ void Camera_PrintSettings(Camera* camera) {
         sp48[i] = '\0';
 
         sp48[camera->play->activeCamId] = 'a';
+#if OOT_DEBUG
         DebugCamera_ScreenTextColored(3, 22, DEBUG_CAM_TEXT_WHITE, sp58);
```
Comment (Dragorn421) [inline src/code/z_camera.c:7285]: surprisingly, `DebugCamera_ScreenTextColored` is in `z_debug.c`, and double surprisingly `z_debug.o` is linked into the retail rom, so you don't need to ifdef it out

**Review by Dragorn421 — COMMENTED** [review]
> a linker error seems more appropriate if you use one of these functions without updating the spec.

agreed

------

I have additional changes locally but we'll see later

**Comment by cadmic** [comment]
> I have additional changes locally but we'll see later

Nice, thanks for verifying. Could you push up your branch so we can incorporate the changes here?

File: src/code/z_camera.c (hunk 1 above)
Comment (cadmic) [reply src/code/z_camera.c]: Huh. Removed these ifdefs

**Comment by cadmic** [comment]
After diffing dragorn's branch, I missed removing z_text, so I've done that now. I think this is good to go now though

**Review by Dragorn421 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Comment by MrCheeze** [comment]
Is this expected after this change?

```bash
$ make VERSION=gc-eu-mq
mips-linux-gnu-ld -T build/gc-eu-mq/oot-gc-eu-mq.ld -T build/gc-eu-mq/undefined_syms.txt --no-check-sections --accept-unknown-input-arch --emit-relocs -Map build/gc-eu-mq/oot-gc-eu-mq.map -o build/gc-eu-mq/oot-gc-eu-mq.elf
mips-linux-gnu-ld: build/gc-eu-mq/src/code/z_moji.o: in function `Moji_DrawChar':
src/code/z_moji.c:62: undefined reference to `gMojiFontTLUTs'
mips-linux-gnu-ld: src/code/z_moji.c:62: undefined reference to `gMojiFontTLUTs'
mips-linux-gnu-ld: build/gc-eu-mq/src/code/z_moji.o: in function `Moji_DrawString':
src/code/z_moji.c:88: undefined reference to `gMojiFontTex'
mips-linux-gnu-ld: src/code/z_moji.c:88: undefined reference to `gMojiFontTex'
make: *** [Makefile:398: build/gc-eu-mq/oot-gc-eu-mq.elf] Error 1
```

**Comment by Dragorn421** [comment]
No, z_moji.o should also be left out for retail, woops

**Pull request merged**